Poker.BE/Poker.BE.API/Controllers/AuthenticationController.cs

```diff
@@ -7,7 +10,69 @@
 
 namespace Poker.BE.API.Controllers
 {
-    public class AuthenticationController : ApiController
-    {
-    }
+	public class AuthenticationController : ApiController
+	{
+		#region Fields
+		private IAuthenticationService service;
+		#endregion
+
+		#region Constructors
+		public AuthenticationController()
+		{
+			service = new Service.Services.AuthenticationService();
+		}
+		#endregion
+
+		#region Methods
+		[HttpPost]
+		public HttpResponseMessage Login(LoginRequest request)
+		{
+			var result = default(LoginResult);
+
+			try
+			{
+				result = service.Login(request);
+			}
+			catch (Exception e)
+			{
+				result.ErrorMessage = e.Message;
+				return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, e);
```

Comment:
## The Good 👍 
you got the idea of what exception should caught where. your controller structure is excellent (as far as i can see here, except auto-formatting you need to fix). returning here `Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ...)` is exactly what you need to do.
## The Bad 👎 
you don't log the error with our `logger`! (the service has logger property) it is very important.
## The Ugly 👎 👎 👎 😭 
you already returning the error message of the exception by the `result.ErrorMessage` to the client (well, that reasonable if you don't have the time to make the message to the client prettier than that). why are you returning the full exception as a HTTP Data response?! this is sooo not good. this how you expose an object containing stack trace to your client - a MEGA no-no like security breach. this is definitely not allowed 🚫 .